lib/Dialect/FIRRTL/Import/FIRParserAsserts.cpp

```diff
@@ -413,24 +413,40 @@ ParseResult circt::firrtl::foldWhenEncodedVerifOp(PrintFOp printOp) {
 
     // TODO: Sanitize the label by replacing whitespace with "_" as done in the
     // Scala impl of ExtractTestCode.
+    ValueRange args;
+    if (printOp.operands().size())
+      args = printOp.operands().drop_front();
+    if (args.size())
+      printOp.emitWarning()
+          << "printf-encoded assertion has format string arguments which may "
```

Comment:
This is going to trigger fairly often, yes?

Reply:
I don't think so. This has to be an assert that is printf-encoded, uses the `assert:`/`assume:`/`cover:`/`assertNotX:`, _and the user has not used the builtin helper utilities to generate this_.

Builtin utilities lock this down and won't generate `printf` with format string arguments, e.g., `assert:` comes from here:
  - https://github.com/chipsalliance/rocket-chip/blob/master/src/main/scala/formal/FormalUtils.scala#L72

There are a few usages of raw `printf`-encoded verification statements that do not use the builtin utilities, but I didn't see any that have format string arguments.

tl;dr: seeing this warning should be extremely rare.